underscore.js

```diff
@@ -490,17 +490,21 @@
   };
 
   // Internal implementation of a recursive `flatten` function.
-  var flatten = function(input, shallow, strict) {
-    var output = [], idx = 0;
+  var flatten = function(input, shallow, strict, _output, _idx) {
+    var output = _output ? _output : [], idx = _idx ? _idx : 0;
     for (var i = 0, length = getLength(input); i < length; i++) {
       var value = input[i];
       if (isArrayLike(value) && (_.isArray(value) || _.isArguments(value))) {
         //flatten current level of array or arguments object
-        if (!shallow) value = flatten(value, shallow, strict);
-        var j = 0, len = value.length;
-        output.length += len;
-        while (j < len) {
-          output[idx++] = value[j++];
+        if (!shallow) {
+          flatten(value, shallow, strict, output, idx);
+          idx = output.length;
+        } else {
+          var j = 0, len = value.length;
+          output.length += len;
```

Comment:
Mind dropping this line? It seems we [added](https://github.com/megawac/underscore/commit/63fc82284021b7f7d916b7561d71168935dd60fb#diff-0f36b362a0b81d6f4d4bfd8a7413c75dR476) it a bit prematurely, it's actually a [slowdown](https://github.com/lodash/lodash/pull/1173) in recent node / iojs